abcix/example/kvstore/kvstore.go

```diff
@@ -149,20 +157,19 @@ func (app *Application) CheckBlock(req types.RequestCheckBlock) types.ResponseCh
 	// e.g. "a=41,c=42,alice,100"
 	ret := types.ResponseCheckBlock{}
 
-	lastState := app.state
+	var size int64
 	for _, tx := range req.Txs {
-		newState, gasUsed, err := executeTx(lastState, tx, true)
+		newState, gasUsed, err := executeTx(app.state, tx, true)
```

Comment:
you don't want state to be mutated during `CheckBlock`. state should only be mutated via `DeliverBlock` and `Commit`